server/commands/project_runner.go

```diff
@@ -125,10 +112,22 @@ type DefaultProjectCommandRunner struct {
 	VersionStepRunner          StepRunner
 	RunStepRunner              CustomStepRunner
 	EnvStepRunner              EnvStepRunner
-	WorkingDir                 WorkingDir
+	WorkingDir                 events.WorkingDir
+	WorkingDirLocker           events.WorkingDirLocker
 	Webhooks                   WebhooksSender
-	WorkingDirLocker           WorkingDirLocker
-	AggregateApplyRequirements ApplyRequirement
+	AggregateApplyRequirements events.ApplyRequirement
+}
+
+// WithLocking add project locking functionality to ProjectCommandRunner
+func (p *DefaultProjectCommandRunner) WithLocking(
```

Comment:
kind of weird to introduce the builder pattern for this. I understand it somewhat for data abstractions when the number of `New`s would be too much.  In this case there would only be two different functions, so I'd rather just stick with that for now.